hack/vendor.sh

```diff
@@ -19,7 +19,12 @@ case $# in
 	;;
 # If user passed arguments to the script
 1)
-	eval "$(grep -E "^clone [^ ]+ $1" "$0")"
+	path=$(pwd)"/hack/vendor.sh"
+	if ! cloneGrep="$(grep -E "^clone [^ ]+ $1" $path)"; then
+    	echo >&2 "error: failed to find 'clone ... $1' in $0"
+    	exit 1
```

Comment:
Thanks for your review. @tianon  😄
fixed the space issue.
In addition, I think there is some difference between $path and $BASH_SOURCE. With the $path, we can either execute `./hack/vendor.sh github.com/docker/go-units` or `./vendor.sh github.com/docker/go-units`. While with $BASH_SOURCE, we can only execute `./hack/vendor.sh github.com/docker/go-units`.

Then $path expand the work directory for vendor.sh. Actually at first time, I ran ./vendor.sh, while it does not work for me as the pwd problem.